### PR TITLE
Update create_coco_tf_record.py

### PR DESCRIPTION
**Pull request by sineeli**
# Description
Code breaks when bounding box coordinates  crosses the original image width or height itself. 

* When number of `bbox_annotations` is same as `num_annotations_skipped` in a single example include a check, not to create the `feature_dict` and just return any empty `feature_dict`.
* Later included another check in `tfrecord_lib.py` whether `tf_example` is not empty while writing to output `tfrecord` file.

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Tests

> :memo: Please describe the tests that you ran to verify your changes.
>  
> * Provide instructions so we can reproduce.  
> * Please also list any relevant details for your test configuration.  

**Test Configuration**:

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: official/vision/data/tfrecord_lib.py [hunk 1]
```diff
@@ -168,7 +168,8 @@ def write_tf_record_dataset(output_path, annotation_iterator,
       logging.info('On image %d', idx)
 
     total_num_annotations_skipped += num_annotations_skipped
-    writers[idx % num_shards].write(tf_example.SerializeToString())
+    if tf_example:
```
Comment (luotigerlsx) [inline official/vision/data/tfrecord_lib.py:171]: If `tf_example_iterator` doesn't return anything, the loop simply won't run. Do we really need this `if` condition ?

File: official/vision/data/create_coco_tf_record.py [hunk 2]
```diff
@@ -210,27 +210,30 @@ def bbox_annotations_to_feature_dict(
...
-      'image/object/area':
-          tfrecord_lib.convert_to_feature(data['area'], 'float_list')
-  }
-  if include_masks:
-    feature_dict['image/object/mask'] = (
-        tfrecord_lib.convert_to_feature(data['encoded_mask_png']))
+  feature_dict = {}
+    
+  if not (len(bbox_annotations) == num_skipped):
+      feature_dict = {
```
Comment (luotigerlsx) [inline official/vision/data/create_coco_tf_record.py:216]: The indentation here seems not consistent with the rest.

**Comment by luotigerlsx** [comment]
Hey @sineeli, thanks for the PR. I have added two comments.

File: official/vision/data/create_coco_tf_record.py (hunk 2 above)
Comment (sineeli) [reply official/vision/data/create_coco_tf_record.py]: I have adjusted the indentation @luotigerlsx, please check once

File: official/vision/data/tfrecord_lib.py (hunk 1 above)
Comment (sineeli) [reply official/vision/data/tfrecord_lib.py:171]: yes, `if tf_example` is not required but if in case the feature dict has image meta data but doesn't have any data for bounding box annotations still we are writing to `tfrecords`. If the image doesn't have any bounding box annotations/ skipped due to wrong annotations should we still need to write that image data to output `tfrecords`

Comment (luotigerlsx) [reply official/vision/data/tfrecord_lib.py:171]: As I mentioned, could you help add a test case to cover this edge condition.

File: official/vision/data/create_coco_tf_record.py [hunk 3]
```diff
@@ -352,7 +353,11 @@ def create_tf_example(image,
               encoded_panoptic_masks['instance_mask'])
             })
 
-  example = tf.train.Example(features=tf.train.Features(feature=feature_dict))
+  if feature_dict_len == len(feature_dict):
```
Comment (sineeli) [inline official/vision/data/create_coco_tf_record.py:356]: @luotigerlsx

written this condition to avoid writing a image without any of the object annotations/caption_annotations/panoptic_annotation

File: official/vision/data/tfrecord_lib_test.py [hunk 4]
```diff
@@ -87,7 +88,90 @@ def test_convert_to_feature_bytes_list(self):
 
     proto = tfrecord_lib.convert_to_feature([b'123', b'456'])
     self.assertSequenceAlmostEqual(proto.bytes_list.value, [b'123', b'456'])
+  
+  def test_obj_annotation_tf_example(self):
```
Comment (sineeli) [inline official/vision/data/tfrecord_lib_test.py:92]: @luotigerlsx 

I have added the test case, can you please check and review the same. Really thanks for reviewing the code.

Comment (luotigerlsx) [reply official/vision/data/tfrecord_lib_test.py:92]: Thanks for adding the test case. It looks good to me. But due to the newly added code, the `lint` test doesn't pass. Please check the log the modify accordingly.

Comment (sineeli) [reply official/vision/data/tfrecord_lib_test.py:93]: I made all the corrections which are required from `lint` of point of view. Thanks

**Review by saberkun — APPROVED** [review]
